src/SAL/recovery/recovery_sfs_cluster.c

```diff
@@ -161,44 +161,59 @@ static char *sfs_cluster_create_val(nfs_client_id_t *clientid, size_t *size)
 	return val;
 }
 
-static int sfs_cluster_recovery_init(void)
-{
+static int sfs_start_grace(const char *vip, int event) {
+	assert(event == EVENT_RELEASE_IP || event == EVENT_TAKE_IP);
+
+	nfs_grace_start_t gsp;
+	gsp.event = event;
+	gsp.ipaddr = (char *)vip;
+
 	int ret;
-	uint64_t version = 0;
+	do {
+		ret = nfs_start_grace(&gsp);
+		if (ret == -EAGAIN) {
+			nfs_wait_for_grace_norefs();
+		} else if (ret) {
+			break;
+		}
+	} while (ret);
 
+	return -ret;
+}
+
+static int sfs_cluster_recovery_init(void)
+{
 	if (sfs_cluster_param.sessionid == 0) {
 		LogCrit(COMPONENT_INIT, "sessionid must be a non-zero value");
 		return -1;
 	}
 
 	sessionid = sfs_cluster_param.sessionid;
 
-	ret = sfs_recovery_backend_init(sessionid, &version);
+	sfs_recovery_backend_init(sessionid, sfs_start_grace);
 
 	LogEvent(COMPONENT_INIT,
-		 "creating sfs_cluster recovery database, sessionid: %d, version: %ld, ret: %d",
-		 sessionid, version, ret);
+		 "creating sfs_cluster recovery database, sessionid: %d", sessionid);
 
-	return ret;
+	return 0;
```

Comment:
should return fail if sfs_recovery_backend_init failed since this return value is used by nfs4_recovery_init.